### PR TITLE
GCI-9: Fix bad code practices from Sonar

### PR DESCRIPTION
**Pull request by collinsgeek**
Removed Useless imports.
Added method override for compareTo() and equals() methods.

@robbyoconnor, Sir I squashed both commits as you requested.

### CODE REVIEW

**Comment by robbyoconnor** [comment]
Use feature branches next time -- we may not get to this right away

**Comment by robbyoconnor** [comment]
@dkayiwa

**Comment by robbyoconnor** [comment]
Did you run this?

**Comment by collinsgeek** [comment]
Alright Sir.

**Comment by robbyoconnor** [comment]
Okay, I'm going to wait for someone else to review this -- on first glance it looks okay...but I don't work on this code

**Comment by collinsgeek** [comment]
Alright @robbyoconner, I guess I will have to wait for the review before I move on to another task?

File: api/src/main/java/org/openmrs/FormField.java [hunk 1]
```diff
@@ -250,9 +250,72 @@ public void setId(Integer id) {
...
+	}
+	if (this.getFormFieldId() != null && other.getFormFieldId() == null) {
+		return 1;
+	}
+	if (this.getFormFieldId() == null && other.getFormFieldId() == null) {
+		return 1;
+	}
+			
+	return this.getFormFieldId().compareTo(other.getFormFieldId());
+ 	}
```
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/FormField.java]: This logic seems a bit overly complex to me...

File: api/src/main/java/org/openmrs/PersonAttribute.java [hunk 2]
```diff
@@ -300,8 +300,24 @@ public void voidAttribute(String reason) {
...
+	if ((retValue = OpenmrsUtil.compareWithNullAsLatest(this.getDateCreated(), other.getDateCreated())) != 0) {
+		return retValue;
+	}
+			
+	if ((retValue = OpenmrsUtil.compareWithNullAsGreatest(this.getValue(), other.getValue())) != 0) {
+		return retValue;
+	}
+			
+	return OpenmrsUtil.compareWithNullAsGreatest(this.getPersonAttributeId(), other.getPersonAttributeId());
 	}
```
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/PersonAttribute.java]: LTGM

File: api/src/main/java/org/openmrs/PersonName.java [hunk 3]
```diff
@@ -525,10 +525,6 @@ public Integer getId() {
 	}
 	
 	/**
-	 * TODO: the behavior of this method needs to be controlled by some sort of global property
-	 * because an implementation can define how they want their names to look (which fields to
-	 * show/hide)
-	 *
```
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/PersonName.java]: Why did you delete this?

File: api/src/main/java/org/openmrs/FormField.java (hunk 1 above)
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/FormField.java]: @dkayiwa you got an opinion?

Comment (collinsgeek) [inline api/src/main/java/org/openmrs/FormField.java]: @robbyoconnor  I followed the definition for the Comparable class definition for this method.

File: api/src/main/java/org/openmrs/PersonAttribute.java (hunk 2 above)
Comment (collinsgeek) [inline api/src/main/java/org/openmrs/PersonAttribute.java]: @robbyoconnor Please Sir, what does LTGM mean?

File: api/src/main/java/org/openmrs/PersonName.java (hunk 3 above)
Comment (collinsgeek) [inline api/src/main/java/org/openmrs/PersonName.java]: @robbyoconnor  Sir, I guessed given the update I made to the compareTo() method following the comparable class definition for compare(). This was no longer necessary?

File: api/src/main/java/org/openmrs/FormField.java (hunk 1 above)
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/FormField.java]: Looks fine nevermind.

File: api/src/main/java/org/openmrs/PersonAttribute.java (hunk 2 above)
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/PersonAttribute.java]: LGTM = Looks Good To Me

File: api/src/main/java/org/openmrs/PersonName.java (hunk 3 above)
Comment (robbyoconnor) [inline api/src/main/java/org/openmrs/PersonName.java]: I'm not sure...

File: api/src/main/java/org/openmrs/PersonAttribute.java (hunk 2 above)
Comment (collinsgeek) [inline api/src/main/java/org/openmrs/PersonAttribute.java]: Nice ^_^

**Comment by robbyoconnor** [comment]
I'm going to approve your task **BUT** if any issues pop up that I missed -- fix them.

**Comment by collinsgeek** [comment]
Ok Sir, I will fix any issues @dkayiwa  may have here :-)

**Comment by robbyoconnor** [comment]
Approved as part of [GCI 2015](https://github.com/openmrs/openmrs-core/pull/1648)

**Comment by robbyoconnor** [comment]
Approved as part of [GCI 2015](https://codein.withgoogle.com/dashboard/task-instances/6455408935829504/?sp-page=1)

**Comment by dkayiwa** [comment]
@collinsgeek can you squash the commits into one?

**Comment by collinsgeek** [comment]
Ok Sir @dkayiwa, what should fix? Am I to revert the changes to compareTo and only leave the useless imports?

**Comment by dkayiwa** [comment]
Is there a way you can make sonar happy without removing DefaultComparator?

**Comment by collinsgeek** [comment]
Ok Sir. Let me try

**Comment by collinsgeek** [comment]
Sir @dkayiwa,  Here is the updated commit, 
the Sonar issue with the method override for compareTo() and equals() methods; is a false postive. So I've reverted the changes in these methods.
See this: https://sonar.spring.io/issue/show/e433a566-4217-40b4-bb4b-cc49ab82a1f9?modal=true

**Comment by robbyoconnor** [comment]
Is there a way to suppress some sonar issues?

**Comment by collinsgeek** [comment]
Sir @robbyoconnor  I don't know

**Comment by lluismf** [comment]
What's the problem implementing equals?
